### PR TITLE
WIP,ENH: some coreg features and fixes

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR :

* renames the MRI groupbox from "Transform" to "HEAD <> MRI Transform"
* brings an immediate solution to ICP "spamming" by disabling all the fit buttons
* fixes the range of the x/y/z spin boxes
* correcty updates the plot when the `trans` parameter is set
* fix the x/y/z scaling spin boxes increment
* improve the UX in scale mode "uniform"

Related to https://github.com/mne-tools/mne-python/pull/10117 and https://github.com/mne-tools/mne-python/issues/8833

### CODE REVIEW

**Comment by larsoner** [comment]
> adds a slider for the head surface opacity (it effectively replaces the previous head_transparency and its automatic opacity management)

I don't think we should get rid of some of the opacity management. I think that whenever the fiducials are unlocked, we should go to a fully-opaque state. (It's okay to let the user change it from this, but I doubt they will.) Once they lock the fiducials, it should go back to its previous opacity (or to the default opacity if the UI launched in edit-fiducials mode).

> fixes the range of the x/y/z combo boxes

Make sure you fix the step, too -- I'd go for 1 mm, 1 degree, and 1 percent for the translation, rotation, and scaling, respectively.

Let me know when I should test!

**Comment by GuillaumeFavelier** [comment]
How is it now @larsoner ?

File: mne/gui/tests/test_coreg_gui.py [hunk 1]
```diff
@@ -176,6 +174,8 @@ def test_coreg_gui_pyvista(tmp_path, renderer_interactive_pyvistaqt):
     assert coreg._orient_glyphs
     assert coreg._scale_by_distance
     assert coreg._mark_inside
+    assert np.isclose(
+        coreg._head_opacity, float(config.get('MNE_COREG_HEAD_OPACITY', '')))
```
Comment (larsoner) [inline mne/gui/tests/test_coreg_gui.py:178]:
```suggestion
    assert_allclose(
        coreg._head_opacity, float(config.get('MNE_COREG_HEAD_OPACITY', '0.95')))
```

**Comment by larsoner** [comment]
> correcty updates the plot when the trans parameter is set

The plot, yes:
```
mne coreg -s sample -d subjects -f MEG/sample/sample_audvis_raw.fif --trans MEG/sample/sample_audvis_raw-trans.fif
```
The rX/rY/rZ, no -- they should not all be 1. Did you fix the range of these so that they are always >= 1? I think these actually in practice go from something like -180 to 180, but please do check the range. It could even be -360 to 360 degrees, not sure...

The translations similarly seem fixed to >= 1, when they should be allowed to range at least -100 to 100 (mm), maybe even -500 to 500 just in case someone's MRI has a very weird affine.

The range of 1 to 1000 percent for scaling seems reasonable, though.

**Comment by larsoner** [comment]
In fixing a test I broke flake, so please fix this while fixing the above stuff:
```
Error: ./mne/gui/tests/test_coreg_gui.py:178:80: E501 line too long (81 > 79 characters)
```

**Comment by hoechenberger** [comment]
- The slider behaves in an odd fashion: there appears to be some kind of a huge step in the beginning, before transparency can be adjusted in a smooth fashion (see video below)
- I thought we had discussed not to add a slider (or any transparency adjustments) if we figure that we can live with a fixed opacity level of 95% – am I mistaken here? 

https://user-images.githubusercontent.com/2046265/150295227-c7bbc9ce-3491-41f0-a9e7-2296d02a8047.mov

**Comment by larsoner** [comment]
> am I mistaken here?

I think you voted against a slider but @drammock and I voted for a slider

**Comment by GuillaumeFavelier** [comment]
> The slider behaves in an odd fashion

Hm... I believe this is how vtk handles transparency. I have the same on machine. I would liek to have a smoother transition too

**Comment by larsoner** [comment]
@GuillaumeFavelier should I test again? Did you test locally a bit to make sure these things worked properly?

**Comment by GuillaumeFavelier** [comment]
> Make sure you fix the step, too -- I'd go for 1 mm, 1 degree, and 1 percent for the translation, rotation, and scaling, respectively.

I did not check this yet.

I still have to deprecate `head_transparency` in favor of `head_opacity` (after https://github.com/mne-tools/mne-python/pull/10222)

I also found a bug with `mark_inside` when using scale mode "uniform".

**Comment by hoechenberger** [comment]
@GuillaumeFavelier @larsoner i will also test tonight (currently out for a walk) and can report back!

**Comment by GuillaumeFavelier** [comment]
In the end, I reproduce the issues that I observe on `main` so I updated #8833.

**Comment by GuillaumeFavelier** [comment]
I spent way too much time debugging this one:

>  the parameter spin boxes force 2 updates consecutively even though only one interaction is done

and the only explanation I found is, quoting `JonB` about `Qt DoubleSpin Box value changed slot has called twice`:

> For anyone reading this: This is known behaviour, which I stumbled across a long time ago and it was infuriating.
>
> With a QSpinbox Qt Internal code uses a timer on clicks/presses. If you step through in debugger you exceed the timeout and get bad/double behaviour.
>
> Simplest: change all connect() to QSpinbox click/key to pass Qt::QueuedConnection as last parameter for connection type, not default Qt::DirectConnection. Now you can use debugger breakpoints and all will be well. I wrote a method to connect all my spinboxes like this.

from https://forum.qt.io/topic/82181/qt-doublespin-box-value-changed-slot-has-called-twice/22

I tried using `Qt.ConnectionType.QueuedConnection` with `PyQt5 5.15.6` but it still did not fix the issue locally.

So using this piece of information:

> QSpinbox Qt Internal code uses a timer on clicks/presses

The idea I have in mind would be to adapt the parallel-running worker to set parameters too so that the connected slot finishes before the timer.

Anyway I plan to get the PR ready after #10237 is merged.

**Comment by GuillaumeFavelier** [comment]
I implemented the idea of parallel worker for `set_parameter` in  09eed42adf0932ff7eb4f0b556b5e5686bc8158d.

**Comment by GuillaumeFavelier** [comment]
I noticed a limitation of using only one `queue`: it's not possible to modify parameters after clicking on `Save subject geometry` so I fixed this in dd7c663dc840064bd247211be14a3da53015a323

**Comment by GuillaumeFavelier** [comment]
After extracting some code from it, now this PR:

* renames the MRI groupbox from "Transform" to "HEAD <> MRI Transform"
* brings an immediate solution to ICP "spamming" by disabling all the fit buttons
* fixes the range of the x/y/z spin boxes
* correcty updates the plot when the `trans` parameter is set
* fix the x/y/z scaling spin boxes increment
* improve the UX in scale mode "uniform"

**Comment by GuillaumeFavelier** [comment]
I fixed the conflicts, can you test it @hoechenberger ?

**Comment by GuillaumeFavelier** [comment]
`62ae738` also fixes:

```
Aligning using ICP
Start     median distance:  17.40 mm
  ICP  1  median distance:  12.46 mm
Traceback (most recent call last):
  File "<decorator-gen-563>", line 24, in _redraw
  File "/Users/hoechenberger/Development/mne-python/mne/gui/_coreg.py", line 692, in _redraw
    for key in self._redraws_pending:
RuntimeError: Set changed size during iteration
```

**Comment by GuillaumeFavelier** [comment]
This is ready for review @larsoner, @hoechenberger

**Comment by GuillaumeFavelier** [comment]
@hoechenberger as per your offline review, I added 4ec8b39

**Pull request closed without merging**